tests/test_database.py

```
@@ -0,0 +1,572 @@
+    async def test_context_manager_disconnect_on_exception(
+        self,
+        db_manager: DatabaseManager,
+    ) -> None:
+        """Test context manager disconnects pool on exception."""
+        mock_pool = AsyncMock(spec=asyncpg.Pool)
+
+        with patch("asyncpg.create_pool", new=AsyncMock(return_value=mock_pool)):
+            with pytest.raises(Exception, match="Test error"):
+                async with db_manager as manager:
+                    assert manager.pool is mock_pool
+                    raise Exception("Test error")
```

Ruff (0.14.5)

`pytest.raises()` block should contain a single simple statement

(PT012)